features/graphql/authorization.feature

```diff
@@ -56,3 +56,112 @@ Feature: Authorization checking
     And the response should be in JSON
     And the header "Content-Type" should be equal to "application/json"
     And the JSON node "errors[0].message" should be equal to "Only admins can create a secured dummy."
+
+  @createSchema
+  Scenario: An admin can create a secured resource
+    When I add "Authorization" header equal to "Basic YWRtaW46a2l0dGVu"
+    And I send the following GraphQL request:
+    """
+    mutation {
+      createSecuredDummy(input: {owner: "someone", title: "Hi", description: "Desc"}) {
+        securedDummy {
+          id
+          title
+          owner
+        }
+      }
+    }
+    """
+    Then the response status code should be 200
+    And the response should be in JSON
+    And the header "Content-Type" should be equal to "application/json"
+    And the JSON node "data.createSecuredDummy.securedDummy.owner" should be equal to "someone"
+
+  Scenario: An admin can create another secured resource
+    When I add "Authorization" header equal to "Basic YWRtaW46a2l0dGVu"
+    And I send the following GraphQL request:
+    """
+    mutation {
+      createSecuredDummy(input: {owner: "dunglas", title: "Hi", description: "Desc"}) {
+        securedDummy {
+          id
+          title
+          owner
+        }
+      }
+    }
+    """
+    Then the response status code should be 200
+    And the response should be in JSON
+    And the header "Content-Type" should be equal to "application/json"
+    And the JSON node "data.createSecuredDummy.securedDummy.owner" should be equal to "dunglas"
+
+  Scenario: A user cannot retrieve an item he doesn't own
+    When I add "Authorization" header equal to "Basic ZHVuZ2xhczprZXZpbg=="
+    And I send the following GraphQL request:
+    """
+    {
+      securedDummy(id: "/secured_dummies/1") {
+        owner
+        title
+      }
+    }
+    """
+    Then the response status code should be 200
+    And the response should be in JSON
+    And the header "Content-Type" should be equal to "application/json"
+    And the JSON node "errors[0].message" should be equal to "Access Denied."
+
+  Scenario: A user can retrieve an item he owns
+    When I add "Authorization" header equal to "Basic ZHVuZ2xhczprZXZpbg=="
+    And I send the following GraphQL request:
+    """
+    {
+      securedDummy(id: "/secured_dummies/2") {
+        owner
+        title
+      }
+    }
+    """
+    Then the response status code should be 200
+    And the response should be in JSON
+    And the header "Content-Type" should be equal to "application/json"
+    And the JSON node "data.securedDummy.owner" should be equal to the string "dunglas"
+
+  Scenario: A user can't assign him an item he doesn't own
```

Comment:
```suggestion
  Scenario: A user can't assign to itself an item it doesn't own
```

Reply:
I just copy/paster the sentences in `deny.feature`. I'm not sure we can say "it" for a human anyway, and the current sentence is correct.

Reply:
"them" could be better: https://en.wikipedia.org/wiki/Singular_they

Reply:
My mistake, it's `they` if we don't want to use `he`.
But `him` has to be replaced anyway to have a correct sentence (at least "himself").